client/state/sites/reducer.js

```diff
@@ -135,43 +135,58 @@ export function items( state = {}, action ) {
 		case SITE_SETTINGS_UPDATE:
 		case SITE_SETTINGS_RECEIVE: {
 			const { siteId, settings } = action;
+			const site = state[ siteId ];
 
-			// A site settings update may or may not include the icon property.
-			// If not, we should simply return state unchanged.
-			if ( ! settings.hasOwnProperty( 'site_icon' ) ) {
+			if ( ! site ) {
 				return state;
 			}
 
-			const mediaId = settings.site_icon;
+			let nextSite = { ...site };
 
-			// Similarly, return unchanged if next icon matches current value,
-			// accounting for the fact that a non-existent icon property is
-			// equivalent to setting the media icon as null
-			const site = state[ siteId ];
-			if ( ! site || ( ! site.icon && null === mediaId ) ||
-					( site.icon && site.icon.media_id === mediaId ) ) {
-				return state;
-			}
+			return reduce( [ 'blog_public', 'site_icon' ], ( memo, key ) => {
+				// A site settings update may or may not include the icon or blog_public property.
+				// If not, we should simply return state unchanged.
+				if ( ! settings.hasOwnProperty( key ) ) {
+					return memo;
+				}
 
-			let nextSite;
-			if ( null === mediaId ) {
-				// Unset icon
-				nextSite = omit( site, 'icon' );
-			} else {
-				// Update icon, intentionally removing reference to the URL,
-				// shifting burden of URL lookup to selector
-				nextSite = {
-					...site,
-					icon: {
-						media_id: mediaId
-					}
-				};
-			}
+				switch ( key ) {
+					case 'blog_public':
+						const isPrivate = parseInt( settings.blog_public, 10 ) !== 1;
+						nextSite.is_private = isPrivate;
```

Comment:
I think we'll want to check (and add test case for) whether the value has actually changed. In this case we're always updating the site state even if the value we've received is the same as what we're currently tracking.